wfcGen.py

```diff
@@ -0,0 +1,287 @@
+                elif (self.gp[0]+1)<len(grid[0]):
+                    if grid[self.gp[1]][self.gp[0]+1].collapsed:
+                        self.entropy = grid[self.gp[1]][self.gp[0]+1].nbg["0"]
+                        self.anyFlag = False
+                    else:
+                        return
+            elif (self.gp[0]+1)<len(grid[0]):
+                if grid[self.gp[1]][self.gp[0]+1].collapsed:
+                    self.entropy = grid[self.gp[1]][self.gp[0]+1].nbg["0"]
+                    self.anyFlag = False
+                else:
+                    return
+            else:
+                return
+
+            if grid[self.gp[1]-1][self.gp[0]].collapsed:
+                self.entropy = intersection(self.entropy, grid[self.gp[1]-1][self.gp[0]].nbg["3"])
+                self.anyFlag = False
+            if grid[self.gp[1]][self.gp[0]-1].collapsed:
+                self.entropy = intersection(self.entropy, grid[self.gp[1]][self.gp[0]-1].nbg["2"])
+                self.anyFlag = False
+            if (self.gp[1]+1)<len(grid):
+                if grid[self.gp[1]+1][self.gp[0]].collapsed:
+                    self.entropy = intersection(self.entropy, grid[self.gp[1]+1][self.gp[0]].nbg["1"])
+                    self.anyFlag = False
+            if (self.gp[0]+1)<len(grid[0]):
+                if grid[self.gp[1]][self.gp[0]+1].collapsed:
+                    self.entropy = intersection(self.entropy, grid[self.gp[1]][self.gp[0]+1].nbg["0"])
+                    self.anyFlag = False
+        else:
+            self.entropy = [self.id]
+        
+    def getEntropy(self):
+        if self.collapsed:
+            return -1
+        else:
+            if self.anyFlag:
+                return len(cells.keys())
+            else:
+                return len(self.entropy)
+
+class Grid:
+    def __init__(self, sx, sy):
+        self.sx = sx
+        self.sy = sy
+        self.entropySave = {}
+        self.grid = []
+        self.last5 = []
+        self.GIMAGE = None
+        self.backUps = []
+        self.backUpsLength = 20
+        self.reloads = 0
+    
+    def setRandom(self):
+        self.grid = []
+        for y in range(self.sy):
+            self.grid.append([])
+            for x in range(self.sx):
+                self.grid[y].append(Cell([x, y]))
+        self.grid[random.randint(0, self.sy-1)][random.randint(0, self.sx-1)].setCell(random.randint(0, len(cells)-1))
+        self.getFinalImage()
+    
+    def loadEntropy(self):
+        self.entropySave = {}
+        for y in range(0, self.sy):
+            for x in range(0, self.sx):
+                if not self.grid[y][x].collapsed:
+                    self.grid[y][x].loadEntropy(self.grid)
+                    if self.grid[y][x].getEntropy() in self.entropySave.keys():
+                        self.entropySave[self.grid[y][x].getEntropy()].append([x, y])
+                    else:
+                        self.entropySave[self.grid[y][x].getEntropy()] = [[x, y]]

```

Comment:
Firstly, notice how you're calling `self.grid[y][x].getEntropy()` several times. Instead, you could save it into a variable.

Secondly, I would suggest that you use the `dict.get` method with a second parameter here.

```suggestion
                    entropy = self.grid[y][x].getEntropy()
                    self.entropySave[entropy] = self.entropySave.get(entropy, []) + [[x, y]]
```

By the way, it is common to store coordinates as tuples (`(x, y)`) rather than lists (`[x, y]`)

Reply:
i'm saving them like that in the case if I would like to change them